backend/tests/e2e/test_forced_conclusion_parallel_e2e.py

```
@@ -274,412 +273,405 @@ async def _execute_forced_conclusion_test(self, test_client, alert_data):
+        # Patch LLM clients (both Gemini SDK and LangChain)
+        with self._create_llm_patch_context(gemini_mock_factory, streaming_mock):
+            # Create a mock initialize method that sets up mock sessions without real server processes
+            async def mock_initialize(self):
+                """Mock initialization that bypasses real server startup."""
+                self.sessions = mock_sessions.copy()
+                self._initialized = True
+                
+            with patch.object(MCPClient, "initialize", mock_initialize), \
+                 patch.object(MCPClient, "list_tools", mock_list_tools), \
+                 patch.object(MCPClient, "call_tool", mock_call_tool):
+                with E2ETestUtils.setup_runbook_service_patching("# Test Runbook\nThis is a test runbook for forced conclusion testing."):
```

Ruff (0.14.11)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)